subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationSpec.groovy

```diff
@@ -332,4 +335,61 @@ class DefaultConfigurationSpec extends Specification {
         dep.taskName == "bar"
         dep.configurationName == "conf"
     }
+
+    def "mutations are prohibited after resolution"() {
+        def listenerBroadcast = Mock(ListenerBroadcast) {
+            getSource() >> Mock(DependencyResolutionListener)
+        }
+        when:
+        def conf = new DefaultConfiguration(":conf", "conf", Mock(ConfigurationsProvider), resolver, listenerManager, Mock(DependencyMetaDataProvider), new DefaultResolutionStrategy())
```

Comment:
Since this comment I've found `UnsupportedConfigurationMutationTest`, and moved the relevant stuff there.